### PR TITLE
PoC of race condition between reload() and cleanTombstones()

### PR DESCRIPTION
**Pull request by ArthurSens**
@roidelapluie 

Adding a test on top of `release-2.25` branch that proves the race condition

### CODE REVIEW

File: tsdb/db_test.go [hunk 1]
```diff
@@ -1143,6 +1143,56 @@ func TestTombstoneCleanFail(t *testing.T) {
...
+
+			db.blocks = append(db.blocks, block)
+		}
+
+		wg.Add(2)
+		// Run reload and CleanTombstones together, with a small time window randomization
+		go func() {
+			defer wg.Done()
+			time.Sleep(time.Duration(rand.Float64() * 100 * float64(time.Millisecond)))
+			require.NoError(t, db.reloadBlocks())
```
Comment (roidelapluie) [inline tsdb/db_test.go:1181]: This is protected by db.cmtx in the minute reloads.

https://github.com/prometheus/prometheus/blob/f0ae5f158428f7c1b3f693c51e3c9d9513a6f6c5/tsdb/db.go#L750-L754

File: tsdb/db_test.go [hunk 2]
```diff
@@ -1143,6 +1143,56 @@ func TestTombstoneCleanFail(t *testing.T) {
...
+		}()
+		go func() {
+			defer wg.Done()
+			time.Sleep(time.Duration(rand.Float64() * 100 * float64(time.Millisecond)))
+			require.NoError(t, db.CleanTombstones())
```
Comment (roidelapluie) [inline tsdb/db_test.go:1186]: This is taking cmtx lock
https://github.com/prometheus/prometheus/blob/f0ae5f158428f7c1b3f693c51e3c9d9513a6f6c5/tsdb/db.go#L1534-L1535

**Review by roidelapluie — COMMENTED** [review]
I don't think there is a race

File: tsdb/db_test.go (hunk 1 above)
Comment (ArthurSens) [reply tsdb/db_test.go:1181]: Yeah, sorry 😬

**Comment by ArthurSens** [comment]
Closing as the test is not representing the real behavior

**Pull request closed without merging**